src/app/home/search/search.component.ts

```diff
@@ -44,7 +77,50 @@ export class SearchComponent {
     });
   }
 
+  ngOnInit(): void {
+    this.subs.add(this.handleFromPlaceValueChanges()).add(this.handleToPlaceValueChanges());
+  }
+
+  ngOnDestroy(): void {
+    this.subs.unsubscribe();
+  }
+
   public onSubmit() {
-    this.homeService.loadTickets(this.searchForm.value);
+    this.homeService.loadTickets(this.searchForm.value).subscribe();
+  }
+
+  private handleFromPlaceValueChanges() {
+    return this.searchForm.controls["fromPlace"].valueChanges
+      .pipe(
+        debounceTime(150),
+        distinctUntilChanged(),
+        mergeMap(value => this.searchService.fetchAirports(value)),
+        map(airports => this.parseAirportsToOptions(airports)),
+        tap(options => {
+          this.fromAirportOptions = options;
+        })
+      )
+      .subscribe();
+  }
+
+  private handleToPlaceValueChanges() {
+    return this.searchForm.controls["toPlace"].valueChanges
+      .pipe(
+        debounceTime(150),
```

Comment:
you can reuse pipe logic by extracting pipe(...) into separate method and calling it inside the pipe() like
`this.searchForm.controls["toPlace"].valueChanges.pipe(fetchAirports())`